app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHours.kt

```diff
@@ -166,7 +167,7 @@ class AddOpeningHours(
     override fun applyAnswerTo(answer: OpeningHoursAnswer, tags: Tags, geometry: ElementGeometry, timestampEdited: Long) {
         if (answer is NoOpeningHoursSign) {
             tags["opening_hours:signed"] = "no"
-            tags.updateCheckDateForKey("opening_hours")
+            tags.updateCheckDateForKey("opening_hours:signed")
```

Comment:
If we don't update or delete the opening hours check date here, the opening hours quest will be asked again right away, wouldn't it?

Reply:
If I read `filter by lazy` correctly, `and (!opening_hours or opening_hours older today -1 years)` is within a block that's been ANDed together with L120, that is only true when OH:signed != no OR when OH:signed = no AND OH:signed is older than a year, which will NOT be true, if we just set OH:signed = no.

The quest would therefore not be asked, even if we came to that point when OH was older than a year.

At least that's what I was thinking. I might need to add tests to confirm that it works as expected.

Reply:
```
    @Test fun `isApplicableTo returns false if the opening hours check date is older than 1 year, while they are not signed and signed check date is less than 1 year old`() {
        assertFalse(questType.isApplicableTo(node(
            tags = mapOf(
                "shop" to "supermarket",
                "name" to "Supi",
                "opening_hours:signed" to "no",
                "check_date:opening_hours" to RelativeDate(-420f).date.toCheckDateString(),
                "check_date:opening_hours:signed" to RelativeDate(-180f).date.toCheckDateString()
            ),
            timestamp = RelativeDate(-365f).date.toCheckDateString().toCheckDate()?.toEpochMilli()
        )))
    }
```
Correctly asserts false